homeassistant/components/agent_dvr/__init__.py

```diff
@@ -0,0 +1,69 @@
+"""Support for Agent."""
+import logging
+
+from agent.a import Agent
+
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import device_registry as dr
+
+from .const import DOMAIN as AGENT_DOMAIN, SERVER_URL
+
+ATTRIBUTION = "ispyconnect.com"
+DEFAULT_BRAND = "Agent DVR by ispyconnect.com"
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_TIMEOUT = 10
+FORWARDS = ["camera"]
+
+
+async def async_setup(hass, config):
+    """Old way to set up integrations."""
+    return True
+
+
+async def async_setup_entry(hass, config_entry):
+    """Set up the Agent component."""
+    hass.data.setdefault(AGENT_DOMAIN, {})
+
+    server_origin = config_entry.data[SERVER_URL]
+    if not server_origin:
```

Comment:
Since you do this check:
How is it possible that the config entry is incomplete?